include/linux/bpf.h

```diff
@@ -173,6 +185,9 @@ union bpf_attr
     // BPF_LINK_DETACH
     bpf_link_detach_attr_t link_detach; ///< Attributes used by BPF_LINK_DETACH.
 
+    // BPF_PROG_ATTACH
+    bpf_prog_attr_t bpf_prog; ///< Attributes used by BPF_PROG_ATTACH and BPF_PROG_DETACH.
```

Comment:
For cross-plat compat, this should be an unnamed struct, not a named struct.  See how BPF_PROG_LOAD is done at line 143.